packages/x-date-pickers/src/internals/components/PickerFieldUI.tsx

```diff
@@ -260,6 +249,14 @@ export function PickerFieldUI<
       </InputAdornment>
     );
   }
+  // handle the case of showing custom `inputAdornment` for Field components
```

Comment:
This is to handle a case when a field component would have an `inputAdornment` slot.
Technically, the API contract declares support for it... 🤷 
Basically, to handle a `DateField` component instead of `DatePicker` [in this demo](https://mui.com/x/react-date-pickers/custom-opening-button/#add-an-icon-next-to-the-opening-button).